### ISSUE TITLE
from 0.64 to 0.65, zpool upgrade fails

### ISSUE THREAD

**Issue by redmop** [zfsonlinux/zfs#3846]
This succeed on the other two servers I have. They were built the same way at the same time. I realize my pool is pretty full. I am retiring a server, and I am moving around data. Could that be the problem?

```
sudo zpool upgrade -a ; echo $?
This system supports ZFS pool feature flags.

cannot set property for 'rpool': invalid argument for this pool operation
255
```
```
sudo zpool status
  pool: rpool
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
	still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
	the pool may no longer be accessible by software that does not support
	the features. See zpool-features(5) for details.
  scan: scrub repaired 0 in 20h48m with 0 errors on Sat Sep 26 14:21:24 2015
config:

	NAME                                                STATE     READ WRITE CKSUM
	rpool                                               ONLINE       0     0     0
	  raidz1-0                                          ONLINE       0     0     0
	    ata-WDC_WD20EARX-00PASB0_WD-WCAZAE751692-part3  ONLINE       0     0     0
	    ata-WDC_WD20EARX-07PASB0_WD-WCAZAE296051-part3  ONLINE       0     0     0
	    ata-WDC_WD20EARX-07PASB0_WD-WCAZAE352946-part3  ONLINE       0     0     0
	    ata-WDC_WD20EARX-07PASB0_WD-WCAZAE361365-part3  ONLINE       0     0     0

```
```
zpool list
NAME    SIZE  ALLOC   FREE  EXPANDSZ   FRAG    CAP  DEDUP  HEALTH  ALTROOT
rpool  7.25T  6.59T   671G         -    45%    90%  1.00x  ONLINE  -
```

**Comment by redmop** [comment]
```
NAME   PROPERTY                    VALUE                       SOURCE
rpool  size                        7.25T                       -
rpool  capacity                    90%                         -
rpool  altroot                     -                           default
rpool  health                      ONLINE                      -
rpool  guid                        11045970619972320805        default
rpool  version                     -                           default
rpool  bootfs                      -                           default
rpool  delegation                  on                          default
rpool  autoreplace                 off                         default
rpool  cachefile                   -                           default
rpool  failmode                    wait                        default
rpool  listsnapshots               off                         default
rpool  autoexpand                  off                         default
rpool  dedupditto                  0                           default
rpool  dedupratio                  1.00x                       -
rpool  free                        669G                        -
rpool  allocated                   6.60T                       -
rpool  readonly                    off                         -
rpool  ashift                      12                          local
rpool  comment                     -                           default
rpool  expandsize                  -                           -
rpool  freeing                     0                           default
rpool  fragmentation               45%                         -
rpool  leaked                      0                           default
rpool  feature@async_destroy       enabled                     local
rpool  feature@empty_bpobj         active                      local
rpool  feature@lz4_compress        active                      local
rpool  feature@spacemap_histogram  active                      local
rpool  feature@enabled_txg         active                      local
rpool  feature@hole_birth          active                      local
rpool  feature@extensible_dataset  enabled                     local
rpool  feature@embedded_data       active                      local
rpool  feature@bookmarks           enabled                     local
rpool  feature@filesystem_limits   disabled                    local
rpool  feature@large_blocks        disabled                    local
```

I can try turning on the features manually. This is a personal server, and only for backups at that. I can offer some help with troubleshooting this if you like.

**Comment by redmop** [comment]
This problem seems to have gone with 0.6.5.2-1~trusty. I did also free up some space. I'm not sure which fixed it.

**Comment by behlendorf** [comment]
@redmop thanks for followig up.  It _may_ have been related to your pool capacity because I don't see any other fixes which were added to 0.6.5.2 which would explain this.  Regardless, I'm glad it now works as expected for you.
